scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/LazyTokenIterator.scala

```diff
@@ -107,6 +107,12 @@ private[parsers] class LazyTokenIterator private (
           case Some(RegionParen(_)) if token.is[LeftParen] => prev.tail
           case Some(RegionEnumArtificialMark) if token.is[KwEnum] => prev.tail
           case Some(_: RegionBrace) if token.is[LeftBrace] => prev.tail
+          /* Handle fewer braces and partial function.
+           * RegionArrow will be added first and then we would add RegionIndent
+           * which would cause the RegionArrow not to be removed on `=>`.
+           * If we add RegionIndent, Arrow will be readded.
+           */
+          case Some(RegionArrow) if dialect.allowFewerBraces && token.is[KwCase] => prev.tail
```

Comment:
does the colon add the arrow region? i don't see any arrows before the first case in the tested example...

Reply:
ah, found it. `case` adds the arrow region while `=>` adds the case region :)

Reply:
I also added a bit better explanation of what is happening here.